### PR TITLE
Refactor balance operations

### LINKED ISSUE #1191: refatorar chamadas de balance data no cockpit
### Contexto
Atualmente a chamada `client.balance.data` faz o agrupamento dos dados de forma pouco eficiente. 
Sempre que essa chamada é feito são feitas requisições para:
- balance 
- bulk anticipations
- balance operations
- recipient

Na maioria dos casos apena a requisição de balance operations é necessária, como por exemplo quando uma nova pagina é requisitada na tabela.
As demais requisições so precisam ser executadas quando o recebedor é alterado.

Dessa forma seria interessante quebra a chamada  `client.balance.data` em  chamadas menores para otimizar as chamadas feitas a api.

### Descrição técnica
- quebra a chamada `client.balance.data` em requisições menores para poder otimizar as requisições feitas a api

### Requisitos e Critérios de Aceite
- [x] garantir que as requisições feitas a api só serão feitas quando algum dado relevante a consulta for alterado

### PR DESCRIPTION
**Pull request by Deivis**
Este PR refatorar a interface de balance operations
## Contexto

Este PR possibilita a utilizaçõa de requisoções tanto em `balance_operations` quanto em `payables` para as constultas de dados da tabela de extrato.
## Checklist
- [x] Refatorar a consulta de `balance_operations`
- [x] Ligar a consulta de `payables` a mesma interface
- [x] Ajustar a tela de extrato para não quebrar com a nova interface

## Issues linkadas
- [x] Resolves #1191 

## Como testar?

1 - O bom e velho `yarn test` no `cockpit`
2 - Abrir a página de extrato(a tabela deve estar funcionando normalmente)

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: a14a61c7a296a1427be920686227edbfb56c358a
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

 * Commit: 1635a57ffd1234658daa705463e53309c45e1672
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: a14a61c7a296a1427be920686227edbfb56c358a
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

 * Commit: 1635a57ffd1234658daa705463e53309c45e1672
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: a14a61c7a296a1427be920686227edbfb56c358a
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

 * Commit: 1635a57ffd1234658daa705463e53309c45e1672
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: a14a61c7a296a1427be920686227edbfb56c358a
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

 * Commit: 1635a57ffd1234658daa705463e53309c45e1672
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

File: packages/cockpit/src/balance/operations/operations.test.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+      const expected = [{
+        amount: 123,
+        type: 'payable',
+      }]
+
+      expect(boletoRefundFeeOutcoming(operation)).toEqual(expected)
+    })
+
+    it('should validate if a is a credit operation', () => {
```
Comment (vagnervst) [inline packages/cockpit/src/balance/operations/operations.test.js]:
```suggestion
    it('should validate if is a credit operation', () => {
```

File: packages/cockpit/src/balance/operations/shared.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
+
```
Comment (vagnervst) [inline packages/cockpit/src/balance/operations/shared.js]: :eyes:

File: packages/cockpit/src/balance/operations/operations.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  pathOr(null, ['movement_object', 'installment'])
+)
+
+export const getSourceId = ifElse(
+  pathEq(['movement_object', 'type'], 'inter_recipient'),
+  path(['movement_object', 'source_id']),
+  always(null)
+)
+
+export const getTarget = ifElse(
```
Comment (vagnervst) [inline packages/cockpit/src/balance/operations/operations.js]: Estas funções `getSourceId`, `getTarget` e `getTransactionId` já estão definidas no `shared`, acho que poderia importa-las de lá

File: packages/cockpit/src/balance/operations/shared.test.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+})
+
+describe('Is negative', () => {
+  it('should validate if a obect prop is negative number', () => {
+    const exampleObject = {
+      batata: -1,
+      xuxu: 0,
+    }
+
+    expect(isNegative('batata', exampleObject)).toBe(true)
```
Comment (rsmelo) [inline packages/cockpit/src/balance/operations/shared.test.js]: criar um teste para o retorno false e outro pro true para ficar mais claro

File: packages/cockpit/src/balance/operations/shared.test.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    expect(isNegative('xuxu', exampleObject)).toBe(false)
+  })
+})
+
+describe('Less than zero', () => {
+  it('should validate if a number is less than zero', () => {
```
Comment (rsmelo) [inline packages/cockpit/src/balance/operations/shared.test.js]: acho esse teste desnecessário, visto que os testes de `isNegative` já cobrem esse trecho de codigo. Porem caso queira manter ele acho interessante quebrar em dois testes um pro caso true e outro pra false

File: packages/cockpit/src/balance/operations/shared.test.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+
+// TODO: review and split this assertions by function
+describe('Comon building props', () => {
+  it('should validate if a given prop is nill', () => {
+    const validateOriginalPaymentDate = isNilProp('original_payment_date')
+    const validatePaymentDate = isNilProp('payment_date')
+    let invalid = validateOriginalPaymentDate(movementObjectMock)
+
+    expect(invalid).toBe(true)
```
Comment (rsmelo) [inline packages/cockpit/src/balance/operations/shared.test.js]: esse teste poderia ser dividido em dois um pra testar o resultado true e outro pro false

File: packages/cockpit/src/balance/operations/shared.test.js [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+  })
+
+  it('should get the transaction id correctly from a payable', () => {
+    const transacitonId = getTransactionId(payableMock)
+
+    expect(transacitonId).toBe(6293604)
+  })
+})
+
+describe('Get type', () => {
```
Comment (rsmelo) [inline packages/cockpit/src/balance/operations/shared.test.js]: estes testes não estão validando os as opçõe abaixo do cond de `getType`
```js
[
    both(
      propEq('type', 'refund'),
      pathEq(['movement_object', 'type'], 'boleto')
    ),
    always('boletoRefundFee'),
  ],
[T, path(['movement_object', 'type'])],
```

File: packages/cockpit/src/balance/operations/operations.test.js [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+
+  it('should get the correct installments', () => {
+    const installments = getInstallment(operationMock)
+    const expected = operationExpectedMock.installment
+
+    expect(installments).toEqual(expected)
+  })
+
+  describe('Building outcoming and outgoing', () => {
+    it('should validate if a operation is a chargeback or refund', () => {
```
[the remark this answers is not in the document]
Comment (Deivis) [reply packages/cockpit/src/balance/operations/operations.test.js]: Feito

File: packages/cockpit/src/balance/operations/shared.test.js (hunk 4 above)
Comment (Deivis) [reply packages/cockpit/src/balance/operations/shared.test.js]: Feito

File: packages/cockpit/src/balance/operations/shared.test.js (hunk 6 above)
Comment (Deivis) [reply packages/cockpit/src/balance/operations/shared.test.js]: Feito

File: packages/cockpit/src/balance/operations/shared.test.js (hunk 7 above)
Comment (Deivis) [reply packages/cockpit/src/balance/operations/shared.test.js]: Feito

File: packages/cockpit/src/balance/operations/operations.test.js [hunk 9]
```diff
@@ -0,0 +1,248 @@
+import operations, {
```
[the remark this answers is not in the document]
Comment (Deivis) [reply packages/cockpit/src/balance/operations/operations.test.js]: Apenas as funções mais simples foram ignoradas como na sugestão, a maior parte das funções continuam sendo exportadas e testadas indivialmente para melhor rastreamento de erros.

File: packages/cockpit/src/balance/operations/operations.test.js [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+  describe('Building outcoming and outgoing', () => {
+    it('should validate if a operation is a chargeback or refund', () => {
+      let refundOrChargeback = isRefundOrChargeBack(operationMock)
+
+      expect(refundOrChargeback).toBe(false)
+
+      refundOrChargeback = isRefundOrChargeBack({
+        ...operationMock,
+        movement_object: {
+          type: 'chargeback',
```
[the remark this answers is not in the document]
Comment (Deivis) [reply packages/cockpit/src/balance/operations/operations.test.js]: Feito

File: packages/cockpit/src/balance/operations/operations.test.js [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+      const outcoming = interRecipientTransferOutcoming(operationMock)
+      const expected = [{
+        amount: 30000,
+        type: 'payable',
+      }]
+
+      expect(outcoming).toEqual(expected)
+    })
+
+    it('should create a correct inter-recipient outcoming', () => {
```
Comment (rsmelo) [inline packages/cockpit/src/balance/operations/operations.test.js]:
```suggestion
    it('should create a correct inter-recipient outgoing', () => {
```

File: packages/cockpit/src/balance/operations/shared.js [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+  )
+)
+
+export const getSourceId = ifElse(
+  pathEq(['movement_object', 'type'], 'inter_recipient'),
+  path(['movement_object', 'source_id']),
+  always(null)
+)
+
+export const getTarget = ifElse(
```
Comment (rsmelo) [inline packages/cockpit/src/balance/operations/shared.js]:
```suggestion
export const getTargetId = ifElse(
```
seria legal atualizar a nomenclatura  para manter o mesmo padrão de `getSourceId` e `getTransactionId`

File: packages/cockpit/src/balance/operations/shared.js [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+export const buildNetAmount = (buildOutcoming, buildOutgoing) => pipe(
+  juxt([
+    pipe(buildOutcoming, pluck('amount'), sum),
+    pipe(buildOutgoing, pluck('amount'), sum),
+  ]),
+  apply(subtract)
+)
+
+export const isNilProp = type => propSatisfies(isNil, type)
```
Comment (rsmelo) [inline packages/cockpit/src/balance/operations/shared.js]: talvez nem precise definir isso como função visto que só usando na linha 55

File: packages/cockpit/src/balance/operations/shared.test.js [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+      amount: 0,
+    }])
+    const builder = buildNetAmount(outcoming, outgoing)
+    const output = builder([])
+
+    expect(output).toEqual(100)
+  })
+})
+
+describe('Is nill prop', () => {
```
Comment (rsmelo) [inline packages/cockpit/src/balance/operations/shared.test.js]: essa função se tornou basicamente um `propSatisfies` do ramda, acho que não tem por que testar ela, pois o teste acaba testando o ramda e não nosso codigo

**Pull request merged**